Update rviz_rendering and rviz_rendering_tests to C++17.

clalancette opened the pull request.

clalancette committed:
Update rviz_rendering and rviz_rendering_tests to C++17.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

clalancette commented:
CI:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=19823)](http://ci.ros2.org/job/ci_linux/19823/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=14323)](http://ci.ros2.org/job/ci_linux-aarch64/14323/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=20544)](http://ci.ros2.org/job/ci_windows/20544/)

mjcarroll reviewed (approved):
I win this one @ahcorde

clalancette committed:
Disable a warning on Windows.

This is because we are using an older version of Eigen
on Windows that deprecated some stuff in C++17.  For
now, just disable this warning.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

clalancette commented:
New CI with the latest changes to solve a warning on Windows:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=19831)](http://ci.ros2.org/job/ci_linux/19831/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=14329)](http://ci.ros2.org/job/ci_linux-aarch64/14329/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=20553)](http://ci.ros2.org/job/ci_windows/20553/)
